### PR TITLE
Support building on Windows with Visual Studio

### PR DESCRIPTION
**Pull request by srcejon**
This PR adds support for building on Windows with Visual Studio.

It updates the root CMakeLists.txt to detect C++11 support in MSVC when CMake >= 3.1 is used.
It prevents GNU/Clang compiler options from being used when the compiler is MSVC.
It removes linking with lib rt, which isn't available on Windows and doesn't appear to be needed.
It ensures symbols are exported in DLLs.

To build on Windows (with CMake >= 3.1):

 mkdir build
 cd build
 cmake ..
 cmake --build . --config Release --target ALL_BUILD

Or better, for a faster build with ninja:

 mkdir build
 cd build
 cmake -G Ninja ..
 ninja 

Changes tested on CentOS 7 (with CMake 2.8) and Mac (with clang) and both build the test apps.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+        else()
+            message(STATUS "Unable to determine C++11 option for compiler ${CMAKE_CXX_COMPILER}.")
+        endif()
+    else ()
+        set (CMAKE_CXX_STANDARD 11)
+    endif ()
+endmacro(use_cxx11)
+
+use_cxx11()
```
Comment (dnwrnr) [inline CMakeLists.txt:22]: Lets just drop this complexity and use this?
```
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,12 +1,29 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 3.1)
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
```
Comment (dnwrnr) [inline CMakeLists.txt:1]: Let bump this a bit ;)
```
cmake_minimum_required(VERSION 3.13.4)
```

**Review by dnwrnr — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (srcejon) [reply CMakeLists.txt]: Sure. I only added that just in case you needed CMake 2.8 support.

File: CMakeLists.txt (hunk 2 above)
Comment (srcejon) [reply CMakeLists.txt]: Ok

**Pull request merged**